### PR TITLE
Send all requests as JSON

### PR DESCRIPTION
**Pull request by scottusrobus**
To allow for greater flexibility with request data this PR changes how we send data to use a content type of JSON with PUT and POST requests as opposed to POST fields. This makes sending objects like the new metadata property for collections and videos easier to deal with.

### CODE REVIEW

**Comment by scottusrobus** [comment]
@ksheurs quick eyes?

**Comment by ksheurs** [comment]
👍

File: lib/ApiResource.php [hunk 1]
```diff
@@ -79,9 +79,19 @@ private static function _request($method, $path, $data = array()) {
     endif;
 
     if ($method === 'POST' || $method === 'PUT'):
-      curl_setopt($curl, CURLOPT_POST, 1);
+      if ($method === 'PUT'):
```
Comment (ksheurs) [inline lib/ApiResource.php:82]: interesting, i've never seen this `:`, `else:`, and `endif;` syntax in php. new?

Comment (scottusrobus) [reply lib/ApiResource.php:82]: i think this syntax has been around as an alternative. maybe not widely used! i always preferred it for some reason.

**Pull request merged**